pymc_marketing/mmm/models/components/lagging.py

```diff
@@ -0,0 +1,313 @@
+from typing import Dict, Optional, Union
+
+import pymc as pm
+
+from pymc_marketing.mmm.transformers import geometric_adstock, weibull_adstock
+from pymc_marketing.mmm.utils import _get_distribution, _validate_model_config
+
+
+class BaseFunction:
+    def __init__(self, max_lagging, model):
+        self.max_lagging = max_lagging
+        self.model = model
```

Comment:
Think that we can drop the model being passed around overall and just assume that the `apply` method is being called within a model context

i.e 
```python
adstock = MyAdstockClass(max_lagging=12)

with pm.Model(...) as model :
    adstock.apply(channel_data)

Reply:
I think this option and the other can coexist together, meaning users can define everything as classes or strings.